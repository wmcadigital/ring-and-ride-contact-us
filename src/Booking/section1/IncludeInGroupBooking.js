import { useEffect } from "react";
import PropTypes from "prop-types";
import { useFormState, useForm } from "react-final-form";

import FormSection from "../../common/FormSection";
import RadioGroup from "../../common/RadioGroup";
import Question from "../../common/Question";
import RadioButton from "../../common/RadioButton";
import FieldError from "../../common/FieldError";
import ProgressIndicator from "../../common/ProgressIndicator";
import { required } from "../../common/validation";
import getSectionPositionInfo from "../getSectionPosition";
import getAboutSectionName from "./getAboutSectionName";

const IncludeInGroupBooking = ({ setIncludeInGroupBooking }) => {
  const stateApi = useFormState();
  const formApi = useForm();
  const { includeInGroupBooking, registrationNo, passengers } = stateApi.values;

  const error = stateApi.submitFailed
    ? stateApi.errors?.includeInGroupBooking
    : null;

  const question = "Do you want to be included in this group booking?";

  useEffect(() => {
    if (includeInGroupBooking) {
      if (includeInGroupBooking === "no") {
        if (
          registrationNo &&
          passengers.find(
            (passenger) => passenger.registrationNo === registrationNo
          )
        ) {
          const passengersFiltered = passengers.filter(
            (passenger) => passenger.registrationNo !== registrationNo
          );
          formApi.mutators.setFormAttribute("passengers", passengersFiltered);
        }
      }
      setIncludeInGroupBooking(includeInGroupBooking);
    }
  }, [includeInGroupBooking]);

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition={getSectionPositionInfo("1", stateApi)}
        sectionName={getAboutSectionName(stateApi)}
      />
      <Question text={question} />
      <RadioGroup error={error}>
        <FieldError text={error} />
        <RadioButton
          key={1}
          label="Yes"
          validation={required}
          value="yes"
          fieldName="includeInGroupBooking"
        />
        <RadioButton
          key={2}
          label="No"
          validation={required}
          value="no"
          fieldName="includeInGroupBooking"
        />
      </RadioGroup>
    </FormSection>
  );
};

IncludeInGroupBooking.propTypes = {
  setIncludeInGroupBooking: PropTypes.func,
};

IncludeInGroupBooking.defaultProps = {
  setIncludeInGroupBooking: () => {},
};

export default IncludeInGroupBooking;
