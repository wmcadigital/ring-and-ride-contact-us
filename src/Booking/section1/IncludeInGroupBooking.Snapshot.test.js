import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import IncludeInGroupBooking from "./IncludeInGroupBooking";

jest.mock("react-final-form");

describe("IncludeInGroupBooking - snapshots", () => {
  it("renders as expected", () => {
    useFormState.mockImplementationOnce(() => ({
      values: { bookingParty: "behalfGroup" },
    }));
    const renderer = createRenderer();

    renderer.render(<IncludeInGroupBooking />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: { bookingParty: "behalfGroup" },
      submitFailed: true,
      errors: { includeInGroupBooking: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<IncludeInGroupBooking />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
