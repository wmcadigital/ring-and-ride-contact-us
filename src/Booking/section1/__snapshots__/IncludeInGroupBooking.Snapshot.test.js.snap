// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncludeInGroupBooking - snapshots renders as expected 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 1 of 5"
  />
  <Question
    text="Do you want to be included in this group booking?"
  />
  <RadioGroup
    error={null}
  >
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="includeInGroupBooking"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="includeInGroupBooking"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;

exports[`IncludeInGroupBooking - snapshots renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 1 of 5"
  />
  <Question
    text="Do you want to be included in this group booking?"
  />
  <RadioGroup
    error="Required"
  >
    <FieldError
      text="Required"
    />
    <RadioButton
      fieldName="includeInGroupBooking"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="includeInGroupBooking"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;
